plugin/network_processor.go

```diff
@@ -0,0 +1,259 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+)
+
+// -----------------------------------------------------------------------------
+// Types
+// -----------------------------------------------------------------------------
+type networkInfo struct {
+	IpAddresses []string
+}
+
+// Common interface for IP configuration processing
+type networkProcessor interface {
+	processIPConfigurations(ctx context.Context, ipConfigs *[]network.InterfaceIPConfiguration) error
+	getPublicIPAddress(ctx context.Context, publicIPRef *network.PublicIPAddress, ipConf *network.InterfaceIPConfiguration) (*string, error)
+	getNetworkInterface(ctx context.Context, interfaceID string) (*network.Interface, error)
+}
+
+// Standard VM network processor
+type vmNetworkProcessor struct {
+	clients       *azureClients
+	netInfo       *networkInfo
+	resourceGroup string
+}
+
+// VMSS network processor
+type vmssNetworkProcessor struct {
+	clients       *azureClients
+	netInfo       *networkInfo
+	resourceGroup string
+	vmssName      string
+	instanceID    string
+	ifName        string
+}
+
+// -----------------------------------------------------------------------------
+// Common Network Processing Logic
+// -----------------------------------------------------------------------------
+func processNetworkProfile(ctx context.Context, interfaces *[]compute.NetworkInterfaceReference,
+	processor networkProcessor) error {
+	if interfaces == nil {
+		return fmt.Errorf("nil network interfaces")
+	}
```

Comment:
Instead of checking if a slice is nil, it's convention to check that the length of the slice is 0. This captures both the case where the slice is nil and when there are no entries in it.
```suggestion
	if len(interfaces) {
		return fmt.Errorf("nil network interfaces")
	}
```